### PR TITLE
Fix a segfault in `podman inspect -l` w/ no containers

### LINKED ISSUE #6472: weird: podman [image|container] inspect -l : panics
```console
$ ./bin/podman inspect -l
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x147a3c9]

goroutine 1 [running]:
github.com/containers/libpod/libpod.(*Container).ID(...)
        libpod/container.go:481
github.com/containers/libpod/pkg/domain/infra/abi.getContainersAndInputByContext(0x20300000000100, 0xc0003ad220, 0x0, 0x1, 0xc0005a2600, 0xc0001739b8, 0x420eee, 0x7f94c3ee67e0, 0x0, 0xb, ...)
        pkg/domain/infra/abi/containers.go:47 +0x499
github.com/containers/libpod/pkg/domain/infra/abi.getContainersByContext(...)
        pkg/domain/infra/abi/containers.go:70
github.com/containers/libpod/pkg/domain/infra/abi.(*ContainerEngine).ContainerInspect(0xc000010698, 0x1cfc9e0, 0xc000042198, 0xc0003ad220, 0x0, 0x1, 0x19c8fdf, 0x4, 0x1, 0x19c84b7, ...)
        pkg/domain/infra/abi/containers.go:337 +0x7b
github.com/containers/libpod/cmd/podman/inspect.(*inspector).inspect(0xc0004ca0f0, 0xc0003ad220, 0x0, 0x1, 0x3, 0xc0004ca0f0)
        cmd/podman/inspect/inspect.go:114 +0x5df
github.com/containers/libpod/cmd/podman/inspect.Inspect(0xc0003ad220, 0x0, 0x1, 0x19c8fdf, 0x4, 0x1, 0x19c84b7, 0x3, 0x1a179fe, 0x4)
        cmd/podman/inspect/inspect.go:44 +0x9b
main.inspectExec(0x29343e0, 0xc0003ad220, 0x0, 0x1, 0x0, 0x0)
        cmd/podman/inspect.go:35 +0x8d
github.com/spf13/cobra.(*Command).execute(0x29343e0, 0xc00003c1a0, 0x1, 0x1, 0x29343e0, 0xc00003c1a0)
        vendor/github.com/spf13/cobra/command.go:838 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0x2934bc0, 0xc000042198, 0x17b9a40, 0x29de598)
        vendor/github.com/spf13/cobra/command.go:943 +0x317
github.com/spf13/cobra.(*Command).Execute(...)
        vendor/github.com/spf13/cobra/command.go:883
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        vendor/github.com/spf13/cobra/command.go:876
main.Execute()
        cmd/podman/root.go:87 +0xec
main.main()
        cmd/podman/main.go:57 +0x168
```
Same with `podman image inspect -l`, `podman container inspect -l`

master @ c4ccd7cbc1509bab6183c47f740cbf2cc4ee0424

### PR DESCRIPTION
**Pull request by mheon**
We also need to rework container/image inspect to be separate, but that can happen in another PR.

Fixes #6472

### CODE REVIEW

**Comment by mheon** [comment]
Adding a test...

**Comment by mheon** [comment]
Test added

**Comment by QiWang19** [comment]
LGTM

File: test/e2e/inspect_test.go [hunk 1]
```diff
@@ -171,4 +171,12 @@ var _ = Describe("Podman inspect", func() {
...
 		Expect(imageData[0].HealthCheck.Test).To(Equal([]string{"CMD-SHELL", "curl -f http://localhost/ || exit 1"}))
 	})
+
+	It("podman inspect --latest with no container fails", func() {
+		SkipIfRemote()
+
+		session := podmanTest.Podman([]string{"inspect", "--latest"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Not(Equal(0)))
+	})
```
Comment (TomSweeneyRedHat) [inline test/e2e/inspect_test.go:181]: Do we have a test to make sure `podman image inspect --latest` fails when there is an image?

**Comment by TomSweeneyRedHat** [comment]
LGTM
one suggestion/question for consideration

**Pull request merged**